src/shims/unix/solarish/foreign_items.rs

```diff
@@ -21,6 +23,32 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
     ) -> InterpResult<'tcx, EmulateItemResult> {
         let this = self.eval_context_mut();
         match link_name.as_str() {
+            // epoll, eventfd
+            "epoll_create1" => {
+                this.assert_target_os("illumos", "gettid");
```

Comment:
The point of this is to prevent these from being exposed on Solarish, I assume?

If yes, fine to keep them, but please add a comment -- and please fix the function name. :)

Reply:
Yes, there's a difference between Solaris and Illumos. The last one implements `epoll` as Linux does, but doesn't include an ability to use it through `syscall`.

Reply:
What if `assert_target_os` would take a link name itself? It should be less error prone then.

Reply:
We call it in many places where that would be awkward.

Reply:
It is rather unusual to have `assert_target_os` in `foreign_items`. In fact we never did that before your PR. So I don't think this is worth it.